# 2.CNN_RNN_sequence_analysis

To use this repository, first download the data [here](https://drive.google.com/file/d/1wPj4M2a_fWqyV9HKUtycKd6dzvrgNeBb/view?usp=sharing) and uncompress the files in this folder. 

In this example, we show how to use CNN and RNN to predict the functionality of non-coding DNA sequences. We use the data from [DeepSEA](https://www.nature.com/articles/nmeth.3547). As described in [DeepSEA](https://www.nature.com/articles/nmeth.3547) and [DanQ](https://www.ncbi.nlm.nih.gov/pubmed/27084946), the human GRCh37 reference genome was segmented into non-overlapping 200-bp bins. The inputs of the deep learning model are the 1000-bp DNA sequences which are centered on the 200-bp bins. In terms of the labels of those sequences, they were generated by collecting profiles from ENCODE and Roadmap Epigenomics data releases, which resulted in a 919 binary vector for each sequence (690 transcription factor binding profiles, 125 DNase I–hypersensitive profiles and 104 histone-mark profiles). To encode the DNA sequence string into a mathematical form which can be fed to the model, we use the one-hot encoding. In terms of the model, because for DNA sequences, not only do the specific motifs matter, but also the interaction between the upstream and downstream motifs also plays important roles in determining the sequence functionality, we combine CNN and RNN, stacking a bi-directional LSTM layer on top of 1D convolutional layers. The original implementation of [DanQ](https://github.com/uci-cbcl/DanQ) requires Theano, which has been discontinued. We reimplemented the idea solely using Keras.

Reference:
* [Predicting effects of noncoding variants with deep learning–based sequence model](https://www.nature.com/articles/nmeth.3547)
* [DanQ: a hybrid convolutional and recurrent deep neural network for quantifying the function of DNA sequences](https://www.ncbi.nlm.nih.gov/pubmed/27084946), https://github.com/uci-cbcl/DanQ